urlc7e4b8c1f2_normchan_craps_simulator_tgz-pJ8-normchan_crapssim_SimulatorJ8/src/normchan/crapssim/simulation/strategy/ProgressiveComeOutHedgeStrategy10.java

```diff
@@ -0,0 +1,103 @@
+package normchan.crapssim.simulation.strategy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Observable;
+
+import normchan.crapssim.engine.Dice;
+import normchan.crapssim.engine.GameManager;
+import normchan.crapssim.engine.bets.AnyCraps;
+import normchan.crapssim.engine.bets.AnySeven;
+import normchan.crapssim.engine.bets.Bet;
+import normchan.crapssim.engine.bets.Come;
+import normchan.crapssim.engine.bets.PassLine;
+import normchan.crapssim.engine.event.BetEvent;
+import normchan.crapssim.engine.event.RollCompleteEvent;
+import normchan.crapssim.engine.event.SessionEvent;
+import normchan.crapssim.simulation.tracker.ResultTracker;
+
+public class ProgressiveComeOutHedgeStrategy10 extends ProgressiveRollStrategy10 {
+	private final GameManager gameManager;
+	private boolean hedgeNeeded = false;
+	private boolean control = false;
+	private List<Come> comeBets = null;
+	private ResultTracker tracker = null;
+	
+	public ProgressiveComeOutHedgeStrategy10(GameManager manager) {
+		super(manager);
+		this.gameManager = manager;
+		player.addObserver(this);
+	}
+
+	public ProgressiveComeOutHedgeStrategy10(GameManager manager, boolean control) {
+		super(manager);
+		this.gameManager = manager;
+		this.control = control;
+		player.addObserver(this);
+	}
+
+	@Override
+	protected void beforeRoll() {
+		super.beforeRoll();
+		if (hedgeNeeded && !control) {
+			addBet(new AnySeven(layout, player, (int)(0.5*unitSize*comeBets.size())));
+			addBet(new AnyCraps(layout, player, comeBets.size()));
+		}
+	}
+
+	@Override
+	public void update(Observable o, Object arg) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.